Extending a class method with prototype assignment confuses the compiler about membership type

This is an issue I encountered when working on  https://github.com/webpack/webpack/pull/6862

### Bug Repository 
See the bug repo for complete code
**[https://github.com/mohsen1/ts-bug-instance-member](https://github.com/mohsen1/ts-bug-instance-member)**

Run `npm test` to run the test

**Expected behavior:**
No Error 
**Actual behavior:**
```
index.js:7:19 - error TS2424: Class 'Ex' defines instance member function 'foo', but extended class 'MyClass' defines it as instance member property.

7 MyClass.prototype.foo = function() {
                    ~~~

```

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -16103,7 +16103,7 @@ namespace ts {
         }
 
         function isMethodLike(symbol: Symbol) {
-            return !!(symbol.flags & SymbolFlags.Method || getCheckFlags(symbol) & CheckFlags.SyntheticMethod);
+            return !!(symbol.flags & SymbolFlags.Method || getCheckFlags(symbol) & CheckFlags.SyntheticMethod || isFunctionLikeDeclaration(getAssignedJavascriptInitializer(symbol.valueDeclaration)));
         }
 
         /**

--- a/src/compiler/checker.ts
+++ b/src/compiler/checker.ts
@@ -16103,7 +16103,9 @@ namespace ts {
         }
 
         function isMethodLike(symbol: Symbol) {
-            return !!(symbol.flags & SymbolFlags.Method || getCheckFlags(symbol) & CheckFlags.SyntheticMethod || isFunctionLikeDeclaration(getAssignedJavascriptInitializer(symbol.valueDeclaration)));
+            return !!(symbol.flags & SymbolFlags.Method ||
+                      getCheckFlags(symbol) & CheckFlags.SyntheticMethod ||
+                      isInJavaScriptFile(symbol.valueDeclaration) && isFunctionLikeDeclaration(getAssignedJavascriptInitializer(symbol.valueDeclaration)));
         }
 
         /**

--- a/tests/baselines/reference/typeFromPropertyAssignment23.symbols
+++ b/tests/baselines/reference/typeFromPropertyAssignment23.symbols
@@ -1,0 +1,24 @@
+=== tests/cases/conformance/salsa/a.js ===
+class Ex {
+>Ex : Symbol(Ex, Decl(a.js, 0, 0))
+
+    foo() {
+>foo : Symbol(Ex.foo, Decl(a.js, 0, 10))
+    }
+}
+
+class MyClass extends Ex {
+>MyClass : Symbol(MyClass, Decl(a.js, 3, 1))
+>Ex : Symbol(Ex, Decl(a.js, 0, 0))
+
+}
+
+// this override should be fine (even if it's a little odd)
+MyClass.prototype.foo = function() {
+>MyClass.prototype.foo : Symbol(MyClass.foo, Decl(a.js, 7, 1))
+>MyClass.prototype : Symbol(MyClass.foo, Decl(a.js, 7, 1))
+>MyClass : Symbol(MyClass, Decl(a.js, 3, 1))
+>prototype : Symbol(MyClass.prototype)
+>foo : Symbol(MyClass.foo, Decl(a.js, 7, 1))
+}
+

--- a/tests/baselines/reference/typeFromPropertyAssignment23.types
+++ b/tests/baselines/reference/typeFromPropertyAssignment23.types
@@ -1,0 +1,26 @@
+=== tests/cases/conformance/salsa/a.js ===
+class Ex {
+>Ex : Ex
+
+    foo() {
+>foo : () => void
+    }
+}
+
+class MyClass extends Ex {
+>MyClass : MyClass
+>Ex : Ex
+
+}
+
+// this override should be fine (even if it's a little odd)
+MyClass.prototype.foo = function() {
+>MyClass.prototype.foo = function() {} : () => void
+>MyClass.prototype.foo : () => void
+>MyClass.prototype : MyClass
+>MyClass : typeof MyClass
+>prototype : MyClass
+>foo : () => void
+>function() {} : () => void
+}
+

--- a/tests/cases/conformance/salsa/typeFromPropertyAssignment23.ts
+++ b/tests/cases/conformance/salsa/typeFromPropertyAssignment23.ts
@@ -1,0 +1,17 @@
+// @noEmit: true
+// @strict: true
+// @checkJs: true
+// @allowJs: true
+// @Filename: a.js
+class Ex {
+    foo() {
+    }
+}
+
+class MyClass extends Ex {
+
+}
+
+// this override should be fine (even if it's a little odd)
+MyClass.prototype.foo = function() {
+}

